### ISSUE TITLE
Challenge 1

### ISSUE THREAD

**Issue by Adyasha28** [scaleracademy/scaler-september-open-source-challenge#274]
name: Adyasha Samal
github_user_name: Adyasha28

**Comment by Adyasha28** [comment]
Forked Repo URL- https://github.com/Adyasha28/scaler-september-open-source-challenge

![Screenshot (57)](https://user-images.githubusercontent.com/78640059/188175811-e03fc753-92d0-403c-a7ec-26389b31ba51.png)

**Comment by Adyasha28** [comment]
![Screenshot (59)](https://user-images.githubusercontent.com/78640059/188282970-6ac8cf16-7974-49cf-9dc1-e3c708bc19bd.png)

**Comment by Adyasha28** [comment]
challenge 4
<img width="689" alt="image" src="https://user-images.githubusercontent.com/78640059/188328227-4f0588db-2b62-46d9-bd41-9973cd5c1b8e.png">

**Comment by Adyasha28** [comment]
![Screenshot (61)](https://user-images.githubusercontent.com/78640059/189534966-b5773b18-3a4e-4071-922c-4c3b34c93060.png)

**Comment by Adyasha28** [comment]
challenge-13
<img width="916" alt="image" src="https://user-images.githubusercontent.com/78640059/189978602-a048efaf-c523-4102-b57c-3f9094c4eba0.png">

**Comment by Adyasha28** [comment]
Challenge-14
![Screenshot (64)](https://user-images.githubusercontent.com/78640059/190230672-f1de5b88-4c7e-4b41-9d28-d1b0d0b4deb3.png)
<img width="412" alt="image" src="https://user-images.githubusercontent.com/78640059/190230688-acd4643c-c19e-41ad-aabf-3a9c09bd923d.png">
![Screenshot (65)](https://user-images.githubusercontent.com/78640059/190230719-a7f4d32e-c53a-4373-b1ec-d7245e0e86ed.png)

**Comment by Adyasha28** [comment]
Challenge 16

![image](https://user-images.githubusercontent.com/78640059/190669373-aebaa548-77d8-4657-bd15-ce4a8f17440c.png)
![image](https://user-images.githubusercontent.com/78640059/190669397-8711676e-53cd-408d-8ce9-2f5b2e8a217f.png)
![image](https://user-images.githubusercontent.com/78640059/190669416-ef23e188-7ebe-4924-a309-d94435a5d1bc.png)

**Comment by Adyasha28** [comment]
Challenge-17

CREATED [git-desktop](https://github.com/Adyasha28/new_repository_2) repo with a new branch [child](https://github.com/Adyasha28/new_repository_2/tree/child) :
![Screenshot (74)](https://user-images.githubusercontent.com/78640059/190870758-a97ee730-a870-4bbf-9586-3346fb5e6345.png)

CREATED [README.md](https://github.com/Adyasha28/new_repository_2/blob/child/README.md) :
![Screenshot (75)](https://user-images.githubusercontent.com/78640059/190870835-41aa7f21-fcbf-42b9-bfa4-504ad9812e12.png)

PUSHED origin and RAISED a [PR](https://github.com/Adyasha28/new_repository_2/commit/6e3491b4bfab44cf9d4228ddb36d39ff20e092c5) from [child](https://github.com/Adyasha28/new_repository_2/tree/child) branch to [main](https://github.com/Adyasha28/new_repository_2/tree/main) :
![image](https://user-images.githubusercontent.com/78640059/190870959-90b76871-9d65-4aaf-b640-a33ca0a9499c.png)

GitHub Repo [git-desktop](https://github.com/Adyasha28/new_repository_2) :
![image](https://user-images.githubusercontent.com/78640059/190871048-b251fff7-6211-400e-8ade-9a20017e2e0c.png)
![image](https://user-images.githubusercontent.com/78640059/190871061-129be5cf-952f-491d-9606-5cb100e77f92.png)

**Comment by Adyasha28** [comment]
Challenge 20:

Project:- Pooja Box Clone(https://github.com/Adyasha28/PoojaBox_Clone)
![image](https://user-images.githubusercontent.com/78640059/191300189-0d798dae-b144-428d-be70-6b439ce59143.png)

Issue link:
![image](https://user-images.githubusercontent.com/78640059/191300549-85a951dc-9b69-417c-844c-da1231a1e2d6.png)

**Comment by Adyasha28** [comment]
Challenge 12:-

<img width="668" alt="image" src="https://user-images.githubusercontent.com/78640059/191538094-7e00862b-3606-4ee4-8c47-4e4787815c11.png">

**Comment by kuraakhilesh8230** [comment]
> Challenge 20:
> 
> Project:- [Pooja Box Clone](https://github.com/users/Adyasha28/projects/2) ![image](https://user-images.githubusercontent.com/78640059/191300189-0d798dae-b144-428d-be70-6b439ce59143.png)
> 
> Issue link: [pooja-box-clone](https://github.com/Adyasha28/PoojaBox_Clone/issues) ![image](https://user-images.githubusercontent.com/78640059/191300549-85a951dc-9b69-417c-844c-da1231a1e2d6.png)

do challenge-20 correctly because all the screen shots should be attached starting from todo to done and in atlast the screen shot should show done with tick marks for all the files

**Comment by Adyasha28** [comment]
Challenge: 23

The repository must have Development(dev), Master(master), and QA(QA) branches on permanent bases. 
Create Temporary Git Branches as per requirements, they can be deleted whenever needed.
For example:- 
•	Bug Fix
•	Hot Fix
•	Feature Branches
•	Experimental Branches
•	WIP branches


Git Branch naming convention for open-source contribution,
•	Start branch name with a group word
•	Use unique id on branch names
•	Use hyphen or slash as separators
•	Git branch with author name
•	Avoid using numbers only
•	Avoid long descriptive names for long-lived branches

Creating a good pull request
•	Give the feature branch a clear name
•	Give the commits and PRs active and descriptive titles
•	Streamline the process of creating a new PR
•	Give the PR a meaningful description
•	Show the functionality visually, whenever possible
•	Review your own PR before you assign it to others

Finding an issue
•	Issues available for community contribution:
•	 The following tags mark issues that are open for community contribution:
	help wanted: Open to participation from the community but not necessarily beginner-friendly
	good first issue: Open to participation from the community and friendly towards new contributors
o	You do not need our permission to work on one of these issues.
o	You may work on an issue labelled good first issue even if it's not your first issue.
•	Issues not available for community contribution:
o	The following tags mark issues that are not open for community contribution:
	🔒 staff only: Requires infrastructure access or institutional knowledge that would be impractical to provide to the community
o	Do not work on these.
•	Issues not ready for work:
o	The following tags mark issues that are not open for community contribution:
	🚧 status: blocked: Blocked by other work that needs to be done first
	🧹 status: ticket work required: Needs additional work before it is ready to be taken up
	🚦 status: awaiting triage: Has not been triaged by a maintainer
o	Do not work on these.
•	Issues without any of the above labels:
o	These issues may (or may not) be open for contribution.
o	Please add a comment asking one of the maintainers to triage the issue and label it as appropriate.

**Comment by Adyasha28** [comment]
Challenge: 23

The repository must have Development(dev), Master(master), and QA(QA) branches on permanent bases. 
Create Temporary Git Branches as per requirements, they can be deleted whenever needed.
For example:- 
•	Bug Fix
•	Hot Fix
•	Feature Branches
•	Experimental Branches
•	WIP branches


Git Branch naming convention for open-source contribution,
•	Start branch name with a group word
•	Use unique id on branch names
•	Use hyphen or slash as separators
•	Git branch with author name
•	Avoid using numbers only
•	Avoid long descriptive names for long-lived branches

Creating a good pull request
•	Give the feature branch a clear name
•	Give the commits and PRs active and descriptive titles
•	Streamline the process of creating a new PR
•	Give the PR a meaningful description
•	Show the functionality visually, whenever possible
•	Review your own PR before you assign it to others

Finding an issue
•	Issues available for community contribution:
•	 The following tags mark issues that are open for community contribution:
	help wanted: Open to participation from the community but not necessarily beginner-friendly
	good first issue: Open to participation from the community and friendly towards new contributors
o	You do not need our permission to work on one of these issues.
o	You may work on an issue labelled good first issue even if it's not your first issue.
•	Issues not available for community contribution:
o	The following tags mark issues that are not open for community contribution:
	🔒 staff only: Requires infrastructure access or institutional knowledge that would be impractical to provide to the community
o	Do not work on these.
•	Issues not ready for work:
o	The following tags mark issues that are not open for community contribution:
	🚧 status: blocked: Blocked by other work that needs to be done first
	🧹 status: ticket work required: Needs additional work before it is ready to be taken up
	🚦 status: awaiting triage: Has not been triaged by a maintainer
o	Do not work on these.
•	Issues without any of the above labels:
o	These issues may (or may not) be open for contribution.
o	Please add a comment asking one of the maintainers to triage the issue and label it as appropriate.

**Comment by kuraakhilesh8230** [comment]
congrats you have completed the milestone-2 successfully and you are good to go🎉

**Comment by Adyasha28** [comment]
Challenge 24:-

Requested to work on this [issue](https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/issues/133)

![image](https://user-images.githubusercontent.com/78640059/192113075-28d6e80f-9cdb-4504-a646-1a8aebcc1f4f.png)

**Comment by Adyasha28** [comment]
Challenge 25:

-Got assigned with an issue: [issue](https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/issues/133)

-Created a [PR](https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/pull/170) for resolving the issue:
<img width="950" alt="image" src="https://user-images.githubusercontent.com/78640059/192159349-285533d1-addf-436e-9e1f-2d9afbf8ad1a.png">

**Comment by Adyasha28** [comment]
Challenge 27:

Some tips for Hacktoberfest-2022:-

- Pull requests can be made in any [GITHUB](https://github.com/topics/hacktoberfest) or [GITLAB](https://gitlab.com/explore/projects/topics/hacktoberfest) hosted project that’s participating in Hacktoberfest.
- We need to submit at least for 4PRs to complete Hacktoberfest.
- Project maintainers must accept our pull/merge requests for them to count toward our total.
- We need to follow the Hacktoberfest guidelines and rules thoroughly.
- Avoid contributing to low quality issues/tasks.

**Comment by kuraakhilesh8230** [comment]
congrats you have completed milestone-3 successfully and you are good to go🎉

**Comment by Adyasha28** [comment]
Challenge 28:

Contributed to these Respositories:-
[first-contributions](https://github.com/firstcontributions/first-contributions/pull/54680)
[kadane's Algorithm](https://github.com/PrakashGatiyala/hackerrank_problems-or-hackerearth_problems/pull/170)
[Adyasha's Card](https://github.com/Syknapse/Contribute-To-This-Project/pull/1698)

**Comment by Adyasha28** [comment]
Challenge 29:

UPVOTED some of the project ideas in the discussion:

[Programming club discussion website](https://github.com/scaleracademy/open-source-projects/discussions/63)
[DSA basket](https://github.com/scaleracademy/open-source-projects/discussions/92)
[Portal for all interview experiences with question approaches, editorials, practice link(company specific)](https://github.com/scaleracademy/open-source-projects/discussions/56)

**Comment by Adyasha28** [comment]
Challenge 30:-

I am thankful to Scalar discord for the Scalar-September-open-source-challenge . I got to learn a lot especially about git and github. Open source was really great. Looking forward for more such opportunities ahead.

**Comment by kuraakhilesh8230** [comment]
congrats you have successfully completed milestone-4 🎊
